packages/dht/src/connection/websocket/WebsocketServerConnection.ts

```diff
@@ -44,20 +41,21 @@ export class ServerWebsocket extends EventEmitter<ConnectionEvents> implements I
         this.socket = socket
     }
 
-    private onMessage(message: Message): void {
-        logger.trace('ServerWebsocket::onMessage')
-        if (message.type === MessageType.UTF8) {
-            logger.debug('Received string Message: ' + message.utf8Data)
-        } else if (message.type === MessageType.BINARY) {
-            logger.trace('Received Binary Message of ' + message.binaryData.length + ' bytes')
-            this.emit('data',
-                new Uint8Array(message.binaryData.buffer, message.binaryData.byteOffset,
-                    message.binaryData.byteLength / Uint8Array.BYTES_PER_ELEMENT))
+    private onMessage(message: WebSocket.RawData, isBinary: boolean): void {
+        if (!isBinary) {
+            logger.trace('Received string Message')
+        } else if (message instanceof Buffer) {
+            logger.trace('Received Buffer Message')
+            this.emit('data', new Uint8Array(message))
+        // If in an Karma / Electron test, use the NodeJS implementation
+        } else if (typeof NodeJsBuffer !== 'undefined' && message instanceof NodeJsBuffer) {
+            logger.trace('Received NodeJsBuffer Message')
+            this.emit('data', new Uint8Array(message))
         }
     }
```

Comment:
Is there a real need for the last else if block (else than logging), just collapse with the `message instanceof Buffer` one?

Reply:
removing the NodeJsBuffer completely as it should not be necessary.